reproducers/ReproducerManagedPtr.cpp

```diff
@@ -0,0 +1,76 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+//
+// care::make_managed falls down when there are some c-style arrays and some
+// care::host_device_ptrs as arguments to the same constructor and implicit
+// casts are disabled.
+//
+// A potential solution would be to provide a chai::ManagedDataSplitter
+// wrapper for arguments where the raw pointer needs to be extracted from
+// host_device_ptrs. To avoid specifying the template parameter, a function
+// called chai::make_managed_data_splitter could be used for template
+// deduction.
+//
+
+#define GPU_ACTIVE
+
+// CARE library headers
+#include "care/config.h"
+#include "care/care.h"
+#include "care/host_device_ptr.h"
+#include "care/PointerTypes.h"
+
+class BaseClass {
+   public:
+      CARE_HOST_DEVICE BaseClass() {}
+      CARE_HOST_DEVICE virtual ~BaseClass() {}
+      CARE_HOST_DEVICE virtual int getData(int index) = 0;
+      CARE_HOST_DEVICE virtual void setData(int index, int value) = 0;
+      CARE_HOST_DEVICE virtual int getManagedData(int index) = 0;
+      CARE_HOST_DEVICE virtual void setManagedData(int index, int value) = 0;
+};
+
+class DerivedClass : public BaseClass {
+   public:
+      CARE_HOST_DEVICE DerivedClass(int* data, care::host_device_ptr<int> managedData)
+      : BaseClass(), m_data(data), m_managedData(managedData) {}
+      CARE_HOST_DEVICE virtual ~DerivedClass() {}
+      CARE_HOST_DEVICE int getData(int index) override { return m_data[index]; }
+      CARE_HOST_DEVICE void setData(int index, int value) override { m_data[index] = value; }
+      CARE_HOST_DEVICE int getManagedData(int index) override { return m_managedData[index]; }
+      CARE_HOST_DEVICE void setManagedData(int index, int value) override { m_managedData[index] = value; }
+
+   private:
+      int* m_data;
+      care::host_device_ptr<int> m_managedData;
+};
+
+
+int main(int, char**) {
+   // Set up data
+   int length = 10;
+   int data[10] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
This is not a big deal, but you can just do int data[10] = {0} and then all ten elements would be 0. Would also work

Reply:
Actually, that pattern usually leads to people misunderstanding what is happening. If I did `int data[10] = {1};` most people would think that it would be all filled with 1s, but actually there would be only one 1 and nine 0s. I believe I could do `int data[10] = {};` but that isn't necessarily clear, either.